app/models/saved_claim.rb

```diff
@@ -91,8 +91,24 @@ def form_must_be_string
   def form_matches_schema
     return unless form_is_string
 
-    JSON::Validator.fully_validate(VetsJsonSchema::SCHEMAS[self.class::FORM], parsed_form).each do |v|
-      errors.add(:form, v.to_s)
+    schema = VetsJsonSchema::SCHEMAS[self.class::FORM]
+
+    schema_errors = JSON::Validator.fully_validate_schema(schema, { errors_as_objects: true })
+    clear_cache = false
+    unless schema_errors.empty?
+      Rails.logger.error('SavedClaim schema failed validation! Attempting to clear cache.', { errors: schema_errors })
+      clear_cache = true
+    end
+
+    validation_errors = JSON::Validator.fully_validate(schema, parsed_form, { errors_as_objects: true, clear_cache: })
```

Comment:
🤔 Do we need `clear_cache` in here? seems like we could lean on the source directly:
```suggestion
    validation_errors = JSON::Validator.fully_validate(schema, parsed_form, { errors_as_objects: true, clear_cache: schema_errors.any?})
```

or alternatively set `clear_cache = schema_errors.any?` up on 97; the conditional reassignment bugs me a little, but honestly it might be mostly a bias towards more Functional Programming practices....